rust/kernel/sync/mutex.rs

```diff
@@ -87,7 +91,9 @@ impl<T> LockIniter for Mutex<T> {
     }
 }
 
-pub struct EmptyGuardContext;
+pub struct EmptyGuardContext {
```

Comment:
EmptyGuardContext is used in other locks as well, so we probably need a different type for mutexes.

Reply:
Oh did not see that (I would have expected it to be in guard.rs if it were used elsewhere)